yt_dlp/extractor/zattoo.py

```diff
@@ -243,7 +294,67 @@ def _make_valid_url(tmpl, host):
 
 
 class ZattooIE(ZattooBaseIE):
-    _VALID_URL_TEMPLATE = r'https?://(?:www\.)?%s/watch/(?P<channel>[^/]+?)/(?P<id>[0-9]+)[^/]+(?:/(?P<recid>[0-9]+))?'
+    _VALID_URL_TEMPLATE = r'''(?x)
+                            https?://(?:www\.)?%s/
+                            (?:
+                                (?:
+                                    vod/movies/([A-Za-z0-9]+)|
+                                    (?:program|watch)/([^/]+)/(\d+)|
+                                    live/([^/]+)
+                                )|
+                                (?:
+                                    .+\?recording=([0-9]+)|
+                                    .+\?movie_id=([A-Za-z0-9]+)|
+                                    .+\?channel=([^&]+)(?:
+                                        &program=(\d+)
+                                    )?
+                                )
+                            )'''
+
+    _VALID_URL = _make_valid_url(_VALID_URL_TEMPLATE, ZattooBaseIE._HOST)
+    _TESTS = [{
+        'url': 'https://zattoo.com/channels/german?channel=srf_zwei',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/guide/german?channel=srf1&program=169860555',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/recordings?recording=193615508',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/tc/ptc_recordings_all_recordings?recording=193615420',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/vod/movies/7521',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/program/daserste/210177916',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/ondemand?movie_id=7521&term_token=9f00f43183269484edde',
+        'only_matching': True,
+    }, {
+        'url': 'https://zattoo.com/live/srf1',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        oid1, cid1, pid1, cid2, record_id, oid2, cid3, pid2 = self._match_valid_url(url).groups()
+        ondemand_id = oid1 or oid2
+        channel_id = cid1 or cid2 or cid3
+        program_id = pid1 or pid2
+        if record_id:
+            return self._extract_record(record_id)
+        elif ondemand_id:
+            return self._extract_ondemand(ondemand_id)
+        elif channel_id:
+            if program_id:
+                return self._extract_video(program_id)
+            return self._extract_live(channel_id)
+
+
+class ZattooOldIE(ZattooBaseIE):
```

Comment:
Do not change the old extractor name (it breaks archive). Rename the new one instead